src/allotropy/parsers/moldev_softmax_pro/softmax_pro_parser.py

```diff
@@ -144,21 +157,23 @@ def _iter_fluorescence_measurement_document(
             msg="Unable to find plate block reads per well.",
         )
 
-        for idx, data_element in enumerate(
-            plate_block.block_data.iter_wavelengths(position)
-        ):
+        for idx, data_element in enumerate(plate_block.iter_data_elements(position)):
             yield FluorescencePointDetectionMeasurementDocumentItems(
-                measurement_identifier=str(uuid.uuid4()),
+                measurement_identifier=data_element.uuid,
                 fluorescence=TRelativeFluorescenceUnit(value=data_element.value),
                 compartment_temperature=(
                     None
                     if data_element.temperature is None
-                    else TQuantityValueDegreeCelsius(data_element.temperature)
+                    else TQuantityValueDegreeCelsius(
+                        ValueEnum.NaN
+                        if math.isnan(data_element.temperature)
+                        else data_element.temperature
```

Comment:
Please create a helper for this with-Nan-value construction and use it here and below.